api/v1alpha1/databasecluster_types.go

```diff
@@ -310,6 +311,14 @@ type Monitoring struct {
 	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
 }
 
+// Sharding are the sharding options. Available only for psmdb
+type Sharding struct {
+	Enabled      bool                 `json:"enabled"`
+	Shards       int32                `json:"shards"`
+	ConfigServer *psmdbv1.ReplsetSpec `json:"configServer,omitempty"`
+	Proxy        *psmdbv1.MongosSpec  `json:"proxy,omitempty"`
```

Comment:
Concerned about exposing the psmdb structs, should we create Everests's own abstraction with only the fields we're going to configure and add the new fields once requested?

Reply:
Why do we need proxy here? We already have a proxy in `.spec.proxy`

Reply:
It's actually the `mongos` section of the sharding, which accroding to @hors is basically proxy. Mongo doesn't have any other proxies. But it's a good question if we can reuse the existing `.spec.proxy` section and just map it to the `.sharding.mongos` under the hood, let me check it.

Reply:
That's how we were doing it in the very beginning of Everest. For some random reason we used to always deploy a sharded cluster with a single replicaset

Reply:
Yes, we can definitely reuse it, we even have this `mongos` type 👍